tools/topology/topology2/platform/intel/sdw-amp-generic.conf

```diff
@@ -349,11 +349,21 @@ IncludeByKey.SDW_AMP_FEEDBACK {
 					Object.Widget.host-copier.1 {
 						stream_name	"amp feedback"
 						pcm_id 3
-						Object.Base.audio_format.1 {
-							# 32 -> 16 bits conversion is done here,
-							# so in_bit_depth is 32 (and out_bit_depth is 16).
-							in_bit_depth	32
-						}
+						num_input_audio_formats 1
```

Comment:
@plbossart you've got a lot of failures in the SDW device tests and it is because you have removed a bunch of supported formats here

Reply:
I am not sure what you are referring to @ranj063, I just split input and output definitions. There are a couple of points where the starting point seems wrong, I don't have a clear picture of how these topologies were reviewed and verified.

Reply:
@plbossart [num_input_audio_formats 6](https://github.com/thesofproject/sof/blob/main/tools/topology/topology2/include/pipelines/cavs/host-gateway-capture.conf#L50) and the supported formats are defined in the host-gateway-capture pipeline. We just overwrite the first supported format here and don't touch other supported formats. When you overwrite `num_input_audio_formats` to 1, you will lose the other 5 supported formats. Same as `num_output_audio_formats`

Reply:
Goodness. We should never ever do this sort of things. 

Unless I am gravely mistaken @bardliao @ranj063 the premise of this topology2 is that this looks object-oriented, so there's a presumption of inheritance of properties. Overriding stuff without even a comment to that effect is really pushing it, no? 

Why do we need to convert anyways?